pkg/bpv7/extension_block_dtlsr.go

```diff
@@ -0,0 +1,141 @@
+// SPDX-FileCopyrightText: 2019, 2021 Markus Sommer
+// SPDX-FileCopyrightText: 2019, 2020 Alvar Penning
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+package bpv7
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/dtn7/cboring"
+)
+
+// DTLSRPeerData contains a peer's connection data
+type DTLSRPeerData struct {
+	// ID is the node's endpoint ID
+	ID EndpointID
+	// Timestamp is the time the last change occurred
+	// when receiving other node's data, we only update if the Timestamp in newer
+	Timestamp DtnTime
+	// Peers is a mapping of previously seen Peers and the respective Timestamp of the last encounter
+	Peers map[EndpointID]DtnTime
+}
+
+func (pd DTLSRPeerData) IsNewerThan(other DTLSRPeerData) bool {
+	return pd.Timestamp > other.Timestamp
+}
+
+// DTLSRBlock contains routing metadata
+//
+// TODO: Turn this into an administrative record
```

Comment:
Would you mind adding a comment marking this `DTLSRBlock` as a non-standard block. As this package represents dtn-bpbis, custom or proprietary resources should be labeled as those. An example could be the custom `SignatureBlock` present in the `pkg/bpv7/extenstion_block_signature.go` file.